Lib/src/main/java/uk/ac/cam/cl/wildpetscience/triton/lib/pipeline/analysis/AnalysisOutputSink.java

```diff
@@ -1,18 +1,140 @@
 package uk.ac.cam.cl.wildpetscience.triton.lib.pipeline.analysis;
 
-import uk.ac.cam.cl.wildpetscience.triton.lib.models.AnimalPosition;
-import uk.ac.cam.cl.wildpetscience.triton.lib.models.Zones;
+import org.opencv.core.Point;
+import org.opencv.core.Rect;
+import uk.ac.cam.cl.wildpetscience.triton.lib.models.*;
 import uk.ac.cam.cl.wildpetscience.triton.lib.pipeline.OutputSink;
 
 import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
 
 /**
- * Analyses location data
+ * Performs the analysis stage of the client-side pipeline.
+ * This includes:
+ *      - Receiving cage and zone data from the config stage.
+ *      - Receiving AnimalPosition data from the processing stage.
+ *      - Removing anomalies from the location data and replacing with
+ *        interpolated data.
+ *      - Classifying location data into zones.
+ *      - Summing time the animal has spent in each zone.
+ *      - Computing an activity metric using the velocity of the animal.
+ *      - (Optional: Computing the most common zone transition cycles of
+ *        the animal.)
  */
 public class AnalysisOutputSink implements OutputSink<AnimalPosition>, Analysis {
+    private double cageWidth;
```

Comment:
Not really sure - whatever is outputted from the config stage? Not sure if it matters too much since the dimensions are at the moment just used to convey width:height ratio, but could be useful for the front end.